packages/flutter/lib/src/material/typography.dart

```diff
@@ -269,11 +296,27 @@ class TextTheme {
 ///  * [ThemeData]
 ///  * <http://material.google.com/style/typography.html>
 class Typography {
-  Typography._();
-
   /// A material design text theme with dark glyphs.
-  static const TextTheme black = const TextTheme._black();
+  final TextTheme black;
 
   /// A material design text theme with light glyphs.
-  static const TextTheme white = const TextTheme._white();
+  final TextTheme white;
+
+  const Typography._(this.black, this.white);
+
+  /// Returns the default typography for the specified platform.
+  static Typography forPlatform(TargetPlatform platform) {
```

Comment:
Thoughts on static `forPlatform` vs a factory ctor with required `TargetPlatform`? The only precedent I could find for a sort-of similar platform-specific ctor was the scroll behaviour delegate, but that's a type that actually gets used as a delegate as opposed to this case, which is just a util lookup.

Another option would be to just add black/white getters.